src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -77,7 +77,10 @@ class TooltipRenderedOnPageBody extends React.PureComponent {
         if (prevProps.text === this.props.text) {
             return;
         }
-        this.updateTooltipTextWidth();
+
+        // Reset the tooltip text width to 0 so that we can measure it again.
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({tooltipTextWidth: 0}, this.updateTooltipTextWidth);
```

Comment:
This lint is correct - the placement of this is not ideal as we're adding infinite re-renders here
Can you find a better placement to reset the tooltip text width?

Reply:
```jsx
      if (prevProps.text === this.props.text) {
            return;
        }
```
this line will prevent the infinite re-rendering , we are only updating tooltipTextWidth , once , when text changes

Reply:
ah good point, missed that